cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -224,11 +232,16 @@ getTxId (ShelleyTxBody era tx _) =
 --
 
 data TxIn = TxIn TxId TxIx
-  deriving (Eq, Ord, Show)
+  deriving (Eq, Generic, Ord, Show)
```

Comment:
We're not generally providing generic instances. Do we need to here? I guess it's for the JSON, but it'd probably be better to give an explicit instance and use an object with named fields "txid" and "txix" rather than a JSON 2-element array.